eox_audit_model/models.py

```diff
@@ -0,0 +1,165 @@
+"""
+Temporary module.
+"""
+import inspect
+import traceback
+from enum import IntEnum
+
+from crum import get_current_request, get_current_user
+from django.db import models
+from jsonfield.fields import JSONField
+
+
+class Status(IntEnum):
+    """
+    Class that defines status for the execution of the hook.
+    """
+    SUCCESS = 1
+    FAIL = 0
+
+    @classmethod
+    def choices(cls):
+        """Returns choices for the class"""
+        return [(key.value, key.name) for key in cls]  # pylint: disable=not-an-iterable, useless-suppression
+
+
+def get_current_domain():
+    """Allow to get the current domain for the request.
+
+    Returns:
+        String: Current domain.
+    """
+    request = get_current_request()
+
+    return request.META['HTTP_HOST'] if request else 'Missing domain.'
+
+
+def get_current_performer():
+    """Allow to get the username for the current user.
+
+    Returns:
+        String: Current username.
+    """
+    user = get_current_user()
+
+    return user.username if user else 'Missing user.'
+
+
+class AuditModel(models.Model):
+    """AuditModel.
+
+    Attributes:
+        action: The name of the action, which initializes the process, for example, api student update.
+        audit_date_stamp: Creation date.
+        input_parameters: The parameter used to execute the audit method.
+        traceback_log: traceback if there is an exception.
+        method_name: Audited method.
+        output_parameters: Values returned by the audited method.
+        performer: Username of the user who is executing the current request.
+        site_domain: Domain get from the current request.
+        status: Success or fail. that depends on audited method.
+
+    Methods:
+        execute_action: Allow to audit the execution for the given method.
+    """
+
+    action = models.CharField(max_length=150)
+    audit_date_stamp = models.DateTimeField(auto_now_add=True, editable=False)
+    input_parameters = JSONField(blank=True)
+    traceback_log = models.TextField(null=True, blank=True)
+    method_name = models.CharField(max_length=150)
+    output_parameters = JSONField(blank=True)
+    performer = models.CharField(max_length=150, default=get_current_performer)
```

Comment:
are we  going to store the "username" or the user_id? is this a foreing key or just a text reference?

Reply:
`username` and this is a text reference